File: Python/SelectionSort.py
# actual funtion
def selectionSort(arr):
    for i in range(0, len(arr)):
        min_idx = i
        for j in range(i+1, len(arr)):
            if (arr[j] < arr[min_idx]):
                min_idx = j
        # swap the values
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr

# funtion calling
arr = [70, 56, 23, 19, 25, 37, 48]
print("Array before sort", arr)
result = selectionSort(arr)
print("Array after selection Sort", result)

File: Java/SelectionSort.java
public class SelectionSort {

    public static void selectionSort(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            int min_idx = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[min_idx]) {
                    min_idx = j;
                }
            }
            int temp = arr[i];
                arr[i] = arr[min_idx];
                arr[min_idx] = temp;
        }
    }

    public static void printArray(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int arr[] = { 7, 4, 10, 11, 16, 20, 5, 1 };

        printArray(arr);
        selectionSort(arr);
        printArray(arr);
    }
}
